keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/plugins/ResourceHandler.kt

```diff
@@ -175,6 +176,6 @@ fun <T : ResourceSpec> Collection<ResourceHandler<*, *>>.supporting(
   find { it.supportedKind.specClass == specClass }
 
 class UnsupportedKind(kind: String) :
-  IllegalStateException("No resource handler supporting \"$kind\" is available") {
+  SystemException("No resource handler supporting \"$kind\" is available") {
```

Comment:
Yeah, this is a tricky one, because it happens in both situations. For example, when reading a resource from the DB, if the `kind` is not recognized, it is thrown, but also when you call the `/diff` endpoint with a bogus kind, it will get thrown. The vast majority of cases where it appears are "system" cases, though. I'd propose we wrap it with a `UserException` in REST controllers.

Reply:
that seems like a great idea — no need to conflate the two contexts/use cases if we don't have to

Reply:
Actually the API case is already covered by our REST exception handler, in which case there's no impact on resource events, so I think we're good:
https://github.com/spinnaker/keel/blob/60b97474b4d6d637ef9eb444557a6dddd0a78d2e/keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/ExceptionHandler.kt#L57-L62